core/src/apps/common/confirm.py

```diff
@@ -47,6 +47,37 @@
         return await ctx.wait(dialog) is CONFIRMED
 
 
+async def info_confirm(
+    ctx: wire.Context,
+    content: ui.Component,
+    info_func: Callable,
+    code: int = ButtonRequestType.Other,
+    confirm: ButtonContent = InfoConfirm.DEFAULT_CONFIRM,
+    confirm_style: ButtonStyleType = InfoConfirm.DEFAULT_CONFIRM_STYLE,
+    cancel: ButtonContent = InfoConfirm.DEFAULT_CANCEL,
+    cancel_style: ButtonStyleType = InfoConfirm.DEFAULT_CANCEL_STYLE,
+    info: ButtonContent = InfoConfirm.DEFAULT_INFO,
+    info_style: ButtonStyleType = InfoConfirm.DEFAULT_INFO_STYLE,
+) -> bool:
+    await ctx.call(ButtonRequest(code=code), ButtonAck)
+
+    dialog = InfoConfirm(
+        content, confirm, confirm_style, cancel, cancel_style, info, info_style
+    )
+
+    while True:
+        if __debug__:
+            result = await ctx.wait(dialog, confirm_signal())
+        else:
+            result = await ctx.wait(dialog)
+
+        if result == INFO:
```

Comment:
this must be `if result is INFO` otherwise this is always false

Reply:
hmm, no, the result is true. but still this should be `result is INFO` to clarify intent

Reply:
The `result` returns one of 

```
CONFIRMED = object()
CANCELLED = object()
INFO = object()
```

These are defined as unrelated standalone new objects. How is `is` any better?

Reply:
done

Reply:
`is` is explicit identity testing, which is what we want here.
`==` also does identity testing in this case, but not necessarily in general or if we change the definition of INFO/CONFIRMED/CANCELLED